#!/usr/bin/env python
PACKAGE = "path_planner"

from dynamic_reconfigure.parameter_generator_catkin import *
#from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

gen.add("laser_frame",   str_t,   0, "Laser frame",  "/laser")

gen.add("smoothing",    double_t,    0, "Smoothing factor for the gaussian kernel", 1.0,  0.01, 5.0)
gen.add("radius",    double_t,    0, "Neighbor search radius", 0.2,  0.001, 1.5)
gen.add("num_threads",    int_t,    0, "Num. of threads", 4,  3, 50)
gen.add("flatness_curvature_threshold",    double_t,    0, "Flatness curvature threshold", 0.2,  0.001, 1.0)
kernel_type_enum = gen.enum([ gen.const("Gaussian",      int_t, 0, "Gaussian"),
                       gen.const("Cosine",     int_t, 1, "Cosine")],
                       "Kernel type")
gen.add("kernel_type",    int_t,    0, "Kernel type", 0,  0, 1, edit_method=kernel_type_enum)

exit(gen.generate(PACKAGE, "path_planner", "NormalEstimationPcl"))
